static/app/components/smartSearchBar/utils.tsx

```diff
@@ -273,17 +273,30 @@ export const shortcuts: Shortcut[] = [
     },
   },
   {
-    text: 'Negate',
+    text: 'Exclude',
     shortcutType: ShortcutType.Negate,
     hotkeys: {
-      actual: ['option+1'],
-      display: 'option+!',
+      actual: 'option+1',
+      display: 'option+1',
     },
     icon: <IconExclamation size="xs" color="gray300" />,
     canRunShortcut: tok => {
-      return tok?.type === Token.Filter;
+      return !!(tok && tok.text[0] !== '!' && tok.type === Token.Filter);
```

Comment:
Can't you check the `negated` property?

Reply:
You'll have to first discriminate the token type.

Reply:
Have to manually cast it to a `TokenResult<Token.Filter>`, seems like it won't do it automatically after `token.type === Token.Filter`

Reply:
uhg, it really __should__ discriminate, something is broken if it's not

Reply:
Found out why, need to type the token argument as `TokenResult<Token>` instead of `TokenResult<any>`

Reply:
👍